### ISSUE TITLE
SA1101 wrongly triggers on local generic functions

### ISSUE THREAD

**Issue by mr-smidge** [DotNetAnalyzers/StyleCopAnalyzers#3035]
# Summary

SA1101 (prefix local calls with this) is wrongly triggered when making a call to a _generic_ local function from _within an instance method_ of a class.  It is not triggered when making a call to a generic local function from within a static method, or a non-generic local function.

# Expected Behaviour

SA1101 is not triggered when making calls to a generic local function from within an instance method.

# Observed Behaviour

SA1101 is triggered.

# Library versions tested:

* StyleCop.Analysers v1.1.118
* StyleCop.Analysers.Unstable v1.2.0.66

# Steps to Reproduce

Try the following minimal program:

```csharp
//------------------------------------------------------------------------------
// <copyright file="Program.cs" company="PlaceholderCompany">
// Copyright (c) PlaceholderCompany. All rights reserved.
// </copyright>
//------------------------------------------------------------------------------

namespace UsesLocalFunction
{
    using System;

    /// <summary>
    /// The <see cref="Program" /> class holds the entry-point of the application.
    /// </summary>
    internal class Program
    {
        /// <summary>
        /// Entry-point of the application.
        /// </summary>
        /// <param name="args">Command-line arguments.</param>
        public static void Main(string[] args)
        {
            SayStaticGenericThings();

            var app = new Program();
            app.SayThings();
            app.SayGenericThings();
        }

        private static void SayStaticGenericThings()
        {
            var narrator = "Static Generic Simon";
            void Say<T>(T phrase) => Console.WriteLine($"{narrator} says, \"{phrase}\"");
            Say("Hello, world!"); // Does not trigger SA1101
        }

        private void SayThings()
        {
            var narrator = "Simon";
            void Say(string phrase) => Console.WriteLine($"{narrator} says, \"{phrase}\"");
            Say("Hello, world!"); // Does not trigger SA1101
        }

        private void SayGenericThings()
        {
            var narrator = "Generic Simon";
            void Say<T>(T phrase) => Console.WriteLine($"{narrator} says, \"{phrase}\"");
            Say("Hello, world!"); // Triggers SA1101!
        }
    }
}
```

**Comment by sharwell** [comment]
Duplicate of #3018
